src/main/java/de/embl/cba/mobie/n5/open_organelle/OpenOrganelleN5S3ImageLoader.java

```diff
@@ -0,0 +1,101 @@
+/*
+ * #%L
+ * BigDataViewer core classes with minimal dependencies
+ * %%
+ * Copyright (C) 2012 - 2016 Tobias Pietzsch, Stephan Saalfeld, Stephan Preibisch,
+ * Jean-Yves Tinevez, HongKee Moon, Johannes Schindelin, Curtis Rueden, John Bogovic
+ * %%
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *
+ * 1. Redistributions of source code must retain the above copyright notice,
+ *    this list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright notice,
+ *    this list of conditions and the following disclaimer in the documentation
+ *    and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ * ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDERS OR CONTRIBUTORS BE
+ * LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ * #L%
+ */
+package de.embl.cba.mobie.n5.open_organelle;
+
+import com.amazonaws.client.builder.AwsClientBuilder;
+import com.amazonaws.services.s3.AmazonS3;
+import com.amazonaws.services.s3.AmazonS3ClientBuilder;
+import de.embl.cba.mobie.n5.S3ImageLoader;
+import mpicbg.spim.data.generic.sequence.AbstractSequenceDescription;
+import org.janelia.saalfeldlab.n5.s3.N5AmazonS3Reader;
+
+import java.io.IOException;
+
+
+// TODO: avoid code duplication!
+//  this is essentially identical to N5S3ImageLoader and N5S3OMEZarrImageLoader
+public class OpenOrganelleN5S3ImageLoader extends OpenOrganelleN5ImageLoader implements S3ImageLoader
+{
+    private final String serviceEndpoint;
+    private final String signingRegion;
+    private final String bucketName;
+    private final String key;
+
+    static class  N5S3ReaderCreator
+    {
+        public N5AmazonS3Reader create(String serviceEndpoint, String signingRegion, String bucketName, String key) throws IOException
+        {
+            final AwsClientBuilder.EndpointConfiguration endpoint = new AwsClientBuilder.EndpointConfiguration( serviceEndpoint, signingRegion );
+
+            final AmazonS3 s3 = AmazonS3ClientBuilder
+                    .standard()
+                    .withPathStyleAccessEnabled( true )
+                    .withEndpointConfiguration( endpoint )
```

Comment:
I had to add the following line here to make it work for me (+ the appropriate imports).
```
                    .withCredentials(new AWSStaticCredentialsProvider( new AnonymousAWSCredentials() ))
```
See comment in the PR for details.

I have also implement a helper class to provide the `AmazonS3` instance without knowledge about the credentials in [imagej-utils](https://github.com/tischi/imagej-utils/blob/master/src/main/java/de/embl/cba/tables/S3Utils.java), see [usage here](https://github.com/mobie/mobie-viewer-fiji/blob/master/src/main/java/de/embl/cba/mobie/n5/N5S3ImageLoader.java#L51-L52). I tried to use it here, but ran into some other issue. Maybe you could see if you can find the issue here and make it work with this helper function.

Reply:
> Thanks for working on this @KateMoreva. I tried it out locally and it works for me after making a change to the credentials provider (see review comments for details); without this I get the following error:
> 
> ```
> Exception in thread "Run$_main" com.amazonaws.services.s3.model.AmazonS3Exception: The authorization header is malformed; the region 'us-west-2' is wrong; expecting 'us-east-1' (Service: Amazon S3; Status Code: 400; Error Code: AuthorizationHeaderMalformed; Request ID: YAF3FXVE31M4VZ3H; S3 Extended Request ID: yJfo3XJQ91CwVKh00+kzB7U9nTQlahixuA4iP5fTETcGPh1rqENn+EhK/R0Ci9S2i82u6F975xo=; Proxy: null), S3 Extended Request ID: yJfo3XJQ91CwVKh00+kzB7U9nTQlahixuA4iP5fTETcGPh1rqENn+EhK/R0Ci9S2i82u6F975xo=
> 	at com.amazonaws.http.AmazonHttpClient$RequestExecutor.handleErrorResponse(AmazonHttpClient.java:1811)
> 	at com.amazonaws.http.AmazonHttpClient$RequestExecutor.handleServiceErrorResponse(AmazonHttpClient.java:1395)
> 	at com.amazonaws.http.AmazonHttpClient$RequestExecutor.executeOneRequest(AmazonHttpClient.java:1371)
> 	at com.amazonaws.http.AmazonHttpClient$RequestExecutor.executeHelper(AmazonHttpClient.java:1145)
> 	at com.amazonaws.http.AmazonHttpClient$RequestExecutor.doExecute(AmazonHttpClient.java:802)
> 	at com.amazonaws.http.AmazonHttpClient$RequestExecutor.executeWithTimer(AmazonHttpClient.java:770)
> 	at com.amazonaws.http.AmazonHttpClient$RequestExecutor.execute(AmazonHttpClient.java:744)
> 	at com.amazonaws.http.AmazonHttpClient$RequestExecutor.access$500(AmazonHttpClient.java:704)
> 	at com.amazonaws.http.AmazonHttpClient$RequestExecutionBuilderImpl.execute(AmazonHttpClient.java:686)
> 	at com.amazonaws.http.AmazonHttpClient.execute(AmazonHttpClient.java:550)
> 	at com.amazonaws.http.AmazonHttpClient.execute(AmazonHttpClient.java:530)
> 	at com.amazonaws.services.s3.AmazonS3Client.invoke(AmazonS3Client.java:5062)
> 	at com.amazonaws.services.s3.AmazonS3Client.invoke(AmazonS3Client.java:5008)
> 	at com.amazonaws.services.s3.AmazonS3Client.getAcl(AmazonS3Client.java:3919)
> 	at com.amazonaws.services.s3.AmazonS3Client.getBucketAcl(AmazonS3Client.java:1244)
> 	at com.amazonaws.services.s3.AmazonS3Client.getBucketAcl(AmazonS3Client.java:1234)
> 	at com.amazonaws.services.s3.AmazonS3Client.doesBucketExistV2(AmazonS3Client.java:1372)
> 	at org.janelia.saalfeldlab.n5.s3.N5AmazonS3Reader.<init>(N5AmazonS3Reader.java:209)
> 	at org.janelia.saalfeldlab.n5.s3.N5AmazonS3Reader.<init>(N5AmazonS3Reader.java:132)
> 	at de.embl.cba.mobie.n5.open_organelle.OpenOrganelleN5S3ImageLoader$N5S3ReaderCreator.create(OpenOrganelleN5S3ImageLoader.java:63)
> 	at de.embl.cba.mobie.n5.open_organelle.OpenOrganelleN5S3ImageLoader.<init>(OpenOrganelleN5S3ImageLoader.java:79)
> 	at de.embl.cba.mobie.n5.S3Reader.readKey(S3Reader.java:36)
> 	at de.embl.cba.mobie.n5.open_organelle.OpenOrganelleS3Reader.showHela(OpenOrganelleS3Reader.java:27)
> 	at de.embl.cba.mobie.n5.open_organelle.OpenOrganelleS3Reader.main(OpenOrganelleS3Reader.java:17)
> Disconnected from the target VM, address: '127.0.0.1:51401', transport: 'socket'
> 
> Process finished with exit code 1
> ```
> 
> I think you said it works for you locally, so maybe you have some default settings in the AWS client that are different and cause this change.

Thank you. This code works for me locally without  AnonymousAWSCredentials.

Reply:
The helper for AmazonS3 does not work for me either. I am still trying to figure out why it is not working.

Reply:
> Thank you. This code works for me locally without AnonymousAWSCredentials.

Yes, I think I know now why: I have some credentials stored in `~/.aws`, and I think if this is the case the aws uses these credentials by default and then fails when trying to authenticate. You probably don't have credentials stored and so it works without specifying it.

> The helper for AmazonS3 does not work for me either. I am still trying to figure out why it is not working.

Ok! Would be great if you could figure it out.